Form: Fix SimpleItem.template docissues (#1069)
DevExpress/devextreme-documentation: api-reference/10 UI Widgets/dxForm/5 Item Types/SimpleItem/template.md

            <dxo-label text="Date of birth"></dxo-label>
            <div *dxTemplate>
                <dx-date-box
                    [(value)]="customer.BirthDate">
                    <dx-validator
                        validationGroup="customerData">
                        <dxi-validation-rule 
                            type="required"
                            message="Date of birth is required">
                        </dxi-validation-rule>
                        <dxi-validation-rule 
                            type="range"
                            [max]="maxDate"
                            message="You must be at least 21 years old">
                        </dxi-validation-rule>
                    </dx-validator>
                </dx-date-box>
            </div>
        </dxi-item>
    </dx-form>

    <!-- tab: app.component.ts -->
    import { Component } from '@angular/core';

    @Component({
        selector: 'app-root',
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.css']
    })
    export class AppComponent {
        customer = {
            Email: "",
            FullName: "",
            BirthDate: null
        };

>         maxDate: Date = new Date().setYear(new Date().getYear() - 21);;

Reviewer: Mistake - double ';;'

Author: @jsdmitry Fixed. Thanks!

            DxRequiredRule,
            DxRangeRule
        },
        data() {
            return {
                customer: {
                    Email: "",
                    FullName: "",
                    BirthDate: null
                },
                maxDate: new Date().setYear(new Date().getYear() - 21);
            }
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React, { useState } from 'react';

    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import Form, {
        SimpleItem,
        Label
    } from 'devextreme-react/form';
    import DateBox from 'devextreme-react/date-box';
    import Validator, {
        RequiredRule,
        RangeRule
    } from 'devextreme-react/validator';

    export default function App() {
        const [customer, setCustomer] = useState({
            Email: "",
            FullName: "",
            BirthDate: null
        });

        const maxDate = new Date().setYear(new Date().getYear() - 21);

        const updateBirthDate = e => {
            setCustomer(prevState => {
                return { ...prevState, BirthDate: e.value };

>             });

Reviewer: You can make simply: ```setCustomer(prevState => ({ ...prevState, BirthDate: e.value }));```

Author: @jsdmitry updated

---

> In Angular, Vue, and React, this template can be used to configure a custom editor more natively than using [editorType](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#editorType) and [editorOptions](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#editorOptions). When you configure a custom editor in the template, consider the following specificities:

Reviewer 2: "more natively" Clarify

Author: ```suggestion
In Angular, Vue, and React, this template can be used instead of [editorType](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#editorType) and [editorOptions](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#editorOptions)  to configure a custom editor. It gives you the advantage of using nested configuration components. When you configure a custom editor in the template, consider the following specificities:
```

Reviewer 2: Ok

- Use two-way binding to bind the custom editor to a [formData](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#formData) field.

- If you use validation, define validation rules in the editor, not in the form item.

> - Set the same [validationGroup](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#validationGroup) as used by the **Form** to ensure that the custom editor is validated simultaneously with other form editors.

Reviewer 2: ```suggestion
- Use the same [validationGroup](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#validationGroup) as the **Form** to ensure the custom editor is validated simultaneously with other form editors.
```

            DxRangeRule
        },
        data() {
            return {
                customer: {
                    Email: "",
                    FullName: "",
                    BirthDate: null
                },
                maxDate: new Date()
            }
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';

    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import Form, {
        SimpleItem,
        Label
    } from 'devextreme-react/form';
    import DateBox from 'devextreme-react/date-box';
    import Validator, {
        RequiredRule,
        RangeRule
    } from 'devextreme-react/validator';

    class App extends React.Component {
        constructor(props) {
            super(props);
            this.state = {
                customer: {
                    Email: "",
                    FullName: "",
                    BirthDate: null
                }
            };
            this.maxDate = new Date();
            this.renderDateBox = this.renderDateBox.bind(this);
            this.updateBirthDate = this.updateBirthDate.bind(this);
        }

>         renderDateBox() {

Reviewer 3: It would be more correct to define it as a separate component.
